src/routes/api/db-migrations/db-migration-utils.ts

```diff
@@ -0,0 +1,101 @@
+import { Request } from "express";
+import fs from "fs";
+import path from "path";
+import { isNodeDev, isNodeTest, isNodeProd, getNodeEnv } from "utils/is-node-env";
+import { getLogger } from "config/logger";
+import { exec as execOrigin } from "child_process";
+import { promisify } from "util";
+
+const exec = promisify(execOrigin);
+const logger = getLogger("DBMigration");
+
+/**
+ *
+ * Functiont to get the parem for db migration.
+ *
+ * Intentially to NOT make this param case insensitive to be more safe
+ *
+ * @returns The common param in req.body `targetScript` to migrate up or down
+ */
+export const getTargetScript = (req: Request) => {
+	const targetScript = (req.body || {}).targetScript;
+	if (!targetScript) {
+		throw {
+			statusCode: 400,
+			message: `"targetScript" is mandatory in the request body, but found none.`
+		};
+	}
+	return targetScript;
+};
+
+/* Make sure the script to migrate up or down is the latest (alphabetically) script in this build.
+ * So the following scenarios will fail.
+ * Person A merge script 1, person B merge scripts 2. Now we CAN NOT migrate anymore. Need to revert one PR first to migrate.
+ *
+ * In short, encourage ppl to over-communicate when it comes to DB migrations.
+ */
+export const validateScriptLocally = async (targetScript: string) => {
+
+	const checkDirSurFix = isNodeTest() ? "-test" : "";
```

Comment:
SurFix?...

I mentioned that in another place: should we rather move it to tests dir?